### PR TITLE
Draw oval and rectangle outlines in bitmap

### PR DESCRIPTION
**Pull request by fsih**
### Resolves
Draw oval and rectangle outlines in bitmap

For reviewer: there is a lot going on here. I have reorganized the logical chunks of works into commits, so reading the last 5 commits one by one should help.

### Proposed Changes
Ovals and rectangles have a filled/outlined toggle

### CODE REVIEW

File: src/components/mode-tools/mode-tools.jsx [hunk 1]
```diff
@@ -234,6 +250,46 @@ const ModeToolsComponent = props => {
...
+                <InputGroup>
+                    <LabeledIconButton
+                        highlighted={!props.fillBitmapShapes}
+                        imgSrc={outlineIcon}
+                        title={props.intl.formatMessage(messages.outlined)}
+                        onClick={props.onOutlineShapes}
+                    />
+                </InputGroup>
+                <InputGroup>
+                    <Label text={props.intl.formatMessage(messages.thickness)}>
```
Comment (paulkaplan) [inline src/components/mode-tools/mode-tools.jsx]: Looks like thickness should only be shown if you are on the not filled state

File: src/helper/bit-tools/oval-tool.js [hunk 2]
```diff
@@ -57,17 +57,49 @@ class OvalTool extends paper.Tool {
      */
     onSelectionChanged (selectedItems) {
         this.boundingBoxTool.onSelectionChanged(selectedItems);
-        if ((!this.oval || !this.oval.parent) &&
+        if ((!this.oval || !this.oval.isInserted()) &&
                 selectedItems && selectedItems.length === 1 && selectedItems[0].shape === 'ellipse') {
             // Infer that an undo occurred and get back the active oval
             this.oval = selectedItems[0];
-        } else if (this.oval && this.oval.parent && !this.oval.selected) {
```
Comment (paulkaplan) [inline src/helper/bit-tools/oval-tool.js:64]: What is the change to the if logic involving `isInserted` do?

Comment (fsih) [reply src/helper/bit-tools/oval-tool.js:64]: I'm not sure how, but I hit an issue where the oval had a parent, but it was a group that had no parent, so it wasn't attached to the view. I discovered isInserted in the API and it's exactly what I wanted, that the item is in the view. I might go through later and change more occurrences of parent !== null to isInserted.

Comment (paulkaplan) [reply src/helper/bit-tools/oval-tool.js:64]: ah cool

**Review by paulkaplan — DISMISSED** [review]
LGTM

File: src/components/mode-tools/mode-tools.jsx (hunk 1 above)
Comment (fsih) [reply src/components/mode-tools/mode-tools.jsx]: Fixed

**Pull request merged**